File: services/core/src/main/java/com/previopls/entity/Usuario.java
package com.previopls.entity;

import com.previopls.entity.enums.RolePapel;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "usuarios", indexes = {
        @Index(name = "ix_usuarios_email", columnList = "email", unique = true)
})
@Getter
@Setter
@NoArgsConstructor
public class Usuario {

    @Id
    @GeneratedValue
    @Column(columnDefinition = "uuid")
    private UUID id;

    @Column(nullable = false, length = 120)
    private String nome;

    @Column(nullable = false, unique = true, length = 180)
    private String email;

    @Column(name = "senha_hash", nullable = false, length = 255)
    private String senhaHash;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private RolePapel papel;

    @Column(name = "criado_em", nullable = false, updatable = false)
    private Instant criadoEm = Instant.now();
}

File: services/gateway/app/models/usuario.py
import enum
import uuid
from datetime import datetime

from sqlalchemy import Column, DateTime, Enum, String
from sqlalchemy.dialects.postgresql import UUID

from app.db.session import Base


class RolePapel(str, enum.Enum):
    CONSULTOR = "consultor"
    ADMIN = "admin"
    ANALISTA = "analista"


class Usuario(Base):
    __tablename__ = "usuarios"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    nome = Column(String(120), nullable=False)
    email = Column(String(180), unique=True, nullable=False, index=True)
    senha_hash = Column(String(255), nullable=False)
    papel = Column(Enum(RolePapel, name="role_papel", values_callable=lambda x: [e.value for e in x]),
                   nullable=False, default=RolePapel.CONSULTOR)
    criado_em = Column(DateTime(timezone=True), default=datetime.utcnow, nullable=False)
